lib/geoengineer/utils/aws_clients.rb

```diff
@@ -1,3 +1,45 @@
+require 'aws-sdk-acm'
+require 'aws-sdk-acmpca'
+require 'aws-sdk-apigateway'
+require 'aws-sdk-waf'
+require 'aws-sdk-ses'
+require 'aws-sdk-s3'
+require 'aws-sdk-redshift'
+require 'aws-sdk-pinpoint'
+require 'aws-sdk-cloudfront'
+require 'aws-sdk-cloudwatch'
+require 'aws-sdk-cloudwatchlogs'
+require 'aws-sdk-cloudwatchevents'
+require 'aws-sdk-elasticloadbalancingv2'
+require 'aws-sdk-dynamodb'
+require 'aws-sdk-elasticache'
+require 'aws-sdk-elasticsearchservice'
+require 'aws-sdk-iam'
+require 'aws-sdk-kinesis'
+require 'aws-sdk-lambda'
+require 'aws-sdk-sns'
+require 'aws-sdk-sqs'
+require 'aws-sdk-kafka'
+require 'aws-sdk-kms'
+require 'aws-sdk-ec2'
+require 'aws-sdk-rds'
+require 'aws-sdk-cloudtrail'
+require 'aws-sdk-route53'
+require 'aws-sdk-route53resolver'
+require 'aws-sdk-dax'
+require 'aws-sdk-efs'
+require 'aws-sdk-states'
+require 'aws-sdk-pinpoint'
+require 'aws-sdk-organizations'
+require 'aws-sdk-autoscaling'
+require 'aws-sdk-emr'
+require 'aws-sdk-codebuild'
+require 'aws-sdk-codedeploy'
+require 'aws-sdk-ecs'
+require 'aws-sdk-directconnect'
+require 'aws-sdk-elasticloadbalancing'
+require 'aws-sdk-firehose'
```

Comment:
When I run just this snippet, I still see a substantial time delay:

```ruby
ruby test  1.84s user 0.92s system 79% cpu 3.490 total
```

What do you think about including these `require` statements in the methods that reference them? For example:

```ruby
  def self.api_gateway(provider = nil)
    require 'aws-sdk-apigateway'
    self.client_cache(
      provider,
      Aws::APIGateway::Client
    )
  end
```

This might result in huge performance benefit when running a plan against a single file that only loads a subset of resources.

Reply:
I can do that, good idea.